src/components/date-picker/day.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import {computed} from 'mobx';
+import {observer} from 'mobx-react';
+
+const styles = require('./date-picker.st.css').default;
+
+export interface DayProps {
+    day: number | string;
+    selected?: boolean;
+    currentDay?: boolean;
+    focused?: boolean;
+    partOfPrevMonth?: boolean;
+    partOfNextMonth?: boolean;
+    dataAutomationId?: string;
+
+    onSelect? (day: number | string): void;
+}
+
+@observer
+export class Day extends React.Component<DayProps, {}> {
+    @computed
+    get styles(): string {
+        if (this.props.focused) {
+            return [styles.calendarItem, styles.day, styles.selectedDay].join(' ');
+        } else if (this.props.selected) {
+            return [styles.calendarItem, styles.day, styles.selectedDay].join(' ');
+        } else if (this.props.currentDay) {
+            return [styles.calendarItem, styles.day, styles.currentDay].join(' ');
+        } else if (this.props.partOfNextMonth || this.props.partOfPrevMonth) {
+            return [styles.calendarItem, styles.greyDay].join(' ');
+        } else {
+            return [styles.calendarItem, styles.day].join(' ');
+        }
+    }
```

Comment:
1. is fact that day could be focused and current or other combinations is covered? 
2. grayDay - is bad name. It should be 'otherMonthDay' or smth like that
3. I think each modifier prop should be represented as separate classname, even if 2 classnames will have same styling, that will help working with this component in projects.
4. does it really have to be mobx computed value?